src/Uno.UI/UI/Xaml/Controls/Button/HyperlinkButton.cs

```diff
@@ -20,24 +20,6 @@ public HyperlinkButton()
 			DefaultStyleKey = typeof(HyperlinkButton);
 		}
 
-		protected override void OnApplyTemplate()
-		{
-			base.OnApplyTemplate();
-
-			// This differs from UWP, where it looks for a template child named "ContentPresenter", 
-			// but ultimately sets the underline on the TextBlock of the first ContentPresenter with a {TemplateBinding Content}.
-			// UWP also doesn't seem to do this in OnApplyTemplate (it's done between the first Measure and the first Arrange).
-			if (GetTemplateChild("ContentPresenter") is ContentPresenter contentPresenter)
-			{
-				// Forces ContentPresenter to materialize its template.
-				contentPresenter.Measure(new Size(0, 0));
-				if (VisualTreeHelper.GetChildrenCount(contentPresenter) == 1 && VisualTreeHelper.GetChild(contentPresenter, 0) is TextBlock textBlock)
```

Comment:
The fix is likely to be different than this. The point is not to unconditionally remove the underlining, but to provide the ability for this scenario to remove the underlining when a custom content is provided. 

It's likely that searching for `ImplicitTextBlock` as the type would be enough instead of `TextBlock` because it's an unmodified ContentControl content template.

Reply:
I'm not sure I understand this. Is the latest commit what you expect?

In UWP, I don't see there is `ImplicitTextBlock`, is this an Uno-specific thing for case like this?

```xml
<HyperlinkButton>Text</HyperlinkButton>
```

Reply:
The hyperlink button contains a content presenter:
https://github.com/unoplatform/uno/blob/2f2377a485e57cd7ac32c7e7d228176ba4528a64/src/Uno.UI/UI/Xaml/Style/Generic/Generic.xaml#L201

If the content is a string, like in your sample, the content presenter will use an `ImplicitTextBlock" to render the text.

I think we need to add a runtime test that validates both scenarios, to ensure that we have the proper underlining defined only when needed. We can drill into the visual tree to find the appropriate `Textblock` and `TextDecorations`.